app/main.dev.js

```diff
@@ -443,11 +451,37 @@ ipcMain.on(Messages.CREATE_PROJECT_REQUEST, async (event, project) => {
           }
           break;
         }
-        // case Constants.ProjectType.CLONE_PROJECT_TYPE: {
-        //   response.error = true;
-        //   response.errorMessage = 'Not yet implemented';
-        //   break;
-        // }
+        // In the CREATE_PROJECT_REQUEST handler, modify the CLONE_PROJECT_TYPE case:
+
+        case Constants.ProjectType.CLONE_PROJECT_TYPE: {
+          try {
+            // Import the directory cloning utilities
+            const { cloneDirectoryStructure, createStatWrapConfig } = require('./utils/DirectoryCloneUtils');
+            
+            // Create the target directory if it doesn't exist
+            if (!fs.existsSync(validationReport.project.path)) {
+              fs.mkdirSync(validationReport.project.path, { recursive: true });
+            }
+            
+            // Clone the directory structure
+            await cloneDirectoryStructure(project.sourceDirectory, validationReport.project.path);
+            
+            // Create the StatWrap configuration
+            await createStatWrapConfig(validationReport.project.path, validationReport.project.id, validationReport.project.name);
+            
+            // Create a marker file to identify this as a cloned project
+            const statWrapDir = path.join(validationReport.project.path, Constants.StatWrapFiles.BASE_FOLDER);
+            if (!fs.existsSync(statWrapDir)) {
+              fs.mkdirSync(statWrapDir, { recursive: true });
+            }
+            fs.writeFileSync(path.join(statWrapDir, 'cloned_project_marker'), 'This is a cloned project');
```

Comment:
In the cloned_project_marker file, could we add some additional information?  Specifically I'm thinking of the project path that this was cloned from.